src/ProgressMeter.jl

```diff
@@ -111,40 +114,52 @@ mutable struct ProgressThresh{T<:Real} <: AbstractProgress
     color::Symbol        # default to green
     output::IO           # output stream into which the progress is written
     numprintedvalues::Int   # num values printed below progress in last iteration
+    offset::Int             # position offset of progress bar (default is 0)
 
     function ProgressThresh{T}(thresh;
                                dt::Real=0.1,
                                desc::AbstractString="Progress: ",
                                color::Symbol=:green,
-                               output::IO=stderr) where T
+                               output::IO=stderr,
+                               offset::Int=0) where T
         tfirst = tlast = time()
         printed = false
-        new{T}(thresh, dt, typemax(T), 0, false, tfirst, tlast, printed, desc, color, output, 0)
+        new{T}(thresh, dt, typemax(T), 0, false, tfirst, tlast, printed, desc, color, output, offset, 0)
     end
 end
 
 ProgressThresh(thresh::Real, dt::Real=0.1, desc::AbstractString="Progress: ",
-         color::Symbol=:green, output::IO=stderr) =
-    ProgressThresh{typeof(thresh)}(thresh, dt=dt, desc=desc, color=color, output=output)
+         color::Symbol=:green, output::IO=stderr,
+         offset::Integer=0) =
+    ProgressThresh{typeof(thresh)}(thresh, dt=dt, desc=desc, color=color, output=output, offset=offset)
 
-ProgressThresh(thresh::Real, desc::AbstractString) = ProgressThresh{typeof(thresh)}(thresh, desc=desc)
+ProgressThresh(thresh::Real, desc::AbstractString; offset::Integer=0) = ProgressThresh{typeof(thresh)}(thresh, desc=desc, offset=offset)
 
 #...length of percentage and ETA string with days is 29 characters
 tty_width(desc) = max(0, displaysize()[2] - (length(desc) + 29))
 
 # update progress display
-function updateProgress!(p::Progress; showvalues = Any[], valuecolor = :blue)
+function updateProgress!(p::Progress; showvalues = Any[], valuecolor = :blue, offset::Integer = p.offset, keep = (offset == 0))
+    p.offset = offset
     t = time()
     if p.counter >= p.n
         if p.counter == p.n && p.printed
             percentage_complete = 100.0 * p.counter / p.n
             bar = barstring(p.barlen, percentage_complete, barglyphs=p.barglyphs)
             dur = durationstring(t-p.tfirst)
             msg = @sprintf "%s%3u%%%s Time: %s" p.desc round(Int, percentage_complete) bar dur
+            print(p.output, "\n" ^ p.offset)
+            print(p.output, "\n" ^ p.numprintedvalues)
```

Comment:
Combine these into `print(p.output, "\n"^(p.offset + p.numprintedvalues)` (it's approximately twice as fast, and easier to see the logic). This also applies several places below.